Distributed tracing 2.9 (#62410)
openshift/openshift-docs: modules/distr-tracing-tempo-config-default.adoc

// Module included in the following assemblies:
//
// * distr_tracing_tempo/distr-tracing-tempo-configuring.adoc

:_content-type: REFERENCE
[id="distr-tracing-tempo-config-default_{context}"]
= Distributed tracing default configuration options

The Tempo custom resource (CR) defines the architecture and settings to be used when creating the {TempoShortName} resources. You can modify these parameters to customize your {TempoShortName} implementation to your business needs.

.Tempo generic YAML example
[source,yaml]
----

> apiVersion: tempotracing.io/v1
> kind: Tempo
> metadata:
>   name: name
> spec:
>   strategy: <deployment_strategy>
>   allInOne:
>     options: {}
>     resources: {}
>   agent:
>     options: {}
>     resources: {}
>   collector:
>     options: {}
>     resources: {}
>   sampling:
>     options: {}
>   storage:
>     type:
>     options: {}
>   query:
>     options: {}
>     resources: {}
>   ingester:
>     options: {}
>     resources: {}
>   options: {}

Reviewer: ```suggestion
apiVersion: tempo.grafana.com/v1alpha1
kind: TempoStack
metadata:
  name: name
spec:
  storage: {}
  resources: {}
  storageSize: 200M
  replicationFactor: 1
  retention: {}
  template:
      distributor:{}
      ingester: {}
      compactor: {}
      querier: {}
      queryFrontend: {}
      gateway: {}
```

----

.Tempo parameters
[options="header"]
|===
|Parameter |Description |Values |Default value

|`apiVersion:`
||API version to use when creating the object.
|`tempotracing.io/v1`
|`tempotracing.io/v1`

|`kind:`
|Defines the kind of Kubernetes object to create.
|`tempo`
|

|`metadata:`
|Data that helps uniquely identify the object, including a `name` string, `UID`, and optional `namespace`.
|
|{product-title} automatically generates the `UID` and completes the `namespace` with the name of the project where the object is created.

|`name:`
|Name for the object.
|The name of your {TempoShortName} instance.
|`tempo-all-in-one-inmemory`

|`spec:`
|Specification for the object to be created.

> |Contains all of the configuration parameters for your {TempoShortName} instance. When a common definition for all Tempo components is required, it is defined under the `spec` node. When the definition relates to an individual component, it is placed under the `spec/<component>` node.

Reviewer: ```suggestion
|Contains all of the configuration parameters for your {TempoShortName} instance. When a common definition for all Tempo components is required, it is defined under the `spec` node. When the definition relates to an individual component, it is placed under the `spec/template/<component>` node.
```

|N/A

> |`strategy:`
> |Tempo deployment strategy
> |`allInOne`, `production`, or `streaming`
> |`allInOne`
> 
> |`allInOne:`
> |Because the `allInOne` image deploys the Agent, Collector, Query, Ingester, and Tempo UI in a single pod, configuration for this deployment must nest component configuration under the `allInOne` parameter.
> |
> |
> 
> |`agent:`
> |Configuration options that define the Agent.
> |
> |
> 
> |`collector:`
> |Configuration options that define the Tempo Collector.
> |
> |
> 
> |`sampling:`
> |Configuration options that define the sampling strategies for tracing.
> |
> |
> 
> |`storage:`
> |Configuration options that define the storage. All storage-related options must be placed under `storage`, rather than under the `allInOne` or other component options.
> |
> |
> 
> |`query:`
> |Configuration options that define the Query service.

Reviewer: ```suggestion
|`resources:`
|Tempo stack assigned resources

|`storageSize:`
|Storage size for ingester PVCs
|
|

|`replicationFactor:`
|Configuration for replication factor
|
|

|`retention:`
|Configuration options retention of the traces.
|
|

|`storage:`
|Configuration options that define the storage. All storage-related options must be placed under `storage`, rather than under the `allInOne` or other component options.
|
|

|`template.distributor:`
|Configuration options that tempo distributor
|
|

|`template.ingester:`
|Configuration options that tempo ingester
|
|

|`template.compactor:`
|Configuration options that tempo compactor
|
|

|`template.querier:`
|Configuration options that tempo querier
|
|

|`template.queryFrontend:`
|Configuration options that tempo query-frontend
|
|

|`template.gateway:`
|Configuration options that tempo gateway
|
|
```

> |
> |
> 
> |`ingester:`
> |Configuration options that define the Ingester service.
> |
> |
> |===
> 

Reviewer: ```suggestion
|===

```

> The following example YAML is the minimum required to create a {TempoName} deployment using the default settings.

Reviewer: ```suggestion
The following example YAML is the minimum required to create a {TempoName} deployment using the default settings, this assumed you have deployed some of the object storage backends and created the secret with credentials for access to such object storage.
```

.Example minimum required dist-tracing-all-in-one.yaml
[source,yaml]
----

> apiVersion: tempotracing.io/v1
> kind: Tempo
> metadata:
>   name: tempo-all-in-one-inmemory

Reviewer: ```suggestion
apiVersion: tempo.grafana.com/v1alpha1
kind: TempoStack
metadata:
  name: simplest
spec:
  storage:
    secret:
      name: minio
      type: s3
  resources:
    total:
      limits:
        memory: 2Gi
        cpu: 2000m
  template:
    queryFrontend:
      jaegerQuery:
        enabled: true
        ingress:
          type: route
```

// Module included in the following assemblies:
//
// * distr_tracing_tempo/distr-tracing-tempo-configuring.adoc

:_content-type: REFERENCE
[id="distr-tracing-tempo-config-default_{context}"]
= Distributed tracing default configuration options

The Tempo custom resource (CR) defines the architecture and settings to be used when creating the {TempoShortName} resources. You can modify these parameters to customize your {TempoShortName} implementation to your business needs.

> .Generic Tempo YAML

Reviewer 2: See [IBM Style](https://www.ibm.com/docs/en/ibm-style?topic=elements-files-directories) suggest following the file type with the word _file_
```suggestion
.Generic Tempo YAML file
```

Author: Thank you. Applying manually.

> ====

Author: Thank you. Fixed.

[source,yaml]
----
apiVersion: tempo.grafana.com/v1alpha1
kind: TempoStack
metadata:
  name: name
spec:
  storage: {}
  resources: {}
  storageSize: 200M
  replicationFactor: 1
  retention: {}
  template:
      distributor:{}
      ingester: {}
      compactor: {}
      querier: {}
      queryFrontend: {}
      gateway: {}
----
====

.Tempo parameters
[options="header"]
|===

> |Parameter |Description |Values |Default value

Author: Thank you. I found where the problem was:
```
||API version to use when creating the object.
```
Fixing.

|N/A

|`resources:`
|Resources assigned to the TempoStack.
|
|

|`storageSize:`
|Storage size for ingester PVCs.
|
|

|`replicationFactor:`
|Configuration for the replication factor.
|
|

|`retention:`
|Configuration options for retention of traces.
|
|

|`storage:`
|Configuration options that define the storage. All storage-related options must be placed under `storage` and not under the `allInOne` or other component options.
|
|

|`template.distributor:`
|Configuration options for the Tempo `distributor`.
|
|

|`template.ingester:`
|Configuration options for the Tempo `ingester`.
|
|

|`template.compactor:`
|Configuration options for the Tempo `compactor`.
|
|

|`template.querier:`
|Configuration options for the Tempo `querier`.
|
|

|`template.queryFrontend:`
|Configuration options for the Tempo `query-frontend`.
|
|

|`template.gateway:`
|Configuration options for the Tempo `gateway`.
|
|
|===

.Minimum required configuration

> ====

Author: Thank you. Removed.
